addons/sale_coupon/models/coupon.py

```diff
@@ -11,6 +11,13 @@ class Coupon(models.Model):
         help="The sales order from which coupon is generated")
     sales_order_id = fields.Many2one('sale.order', 'Used in', readonly=True,
         help="The sales order on which the coupon is applied")
+    partner_id = fields.Many2one(compute='_compute_partner', store=True)
+
+    @api.depends('order_id.partner_id')
+    def _compute_partner(self):
+        for re in self:
+            if not re.partner_id.active:
+                re.partner_id = re.order_id.partner_id
```

Comment:
The coupon should not be created with the public user's partner in the first place.

Reply:
@Williambraecky I don't change the partner of the coupon anymore, but I think of another way that is to compare the partner of the sale order and the partner on the mail that sends the code to the partner
![Screenshot_20220922_131507](https://user-images.githubusercontent.com/71593331/191672086-c898fded-e07d-496a-b786-b4e3a9f0ddd5.png)

Reply:
Don't you think it's a bit excessive to render a whole template just to get the partner?